model-optimizer/mo/graph/port.py

```diff
@@ -320,7 +320,12 @@ def disconnect(self):
                     self.node.graph.remove_edge(self.node.id, port.node.id)
             else:
                 for port in consumer_ports:
-                    self.node.graph.remove_edge(port.node.in_node(port.idx).id, port.node.id)
+                    src_node = port.node.in_node(port.idx).id
+                    dst_node = port.node.id
+                    for key, val in self.node.graph.get_edge_data(src_node, dst_node).items():
+                        if val['in'] == port.idx:
+                            self.node.graph.remove_edge(src_node, dst_node, key=key)
+                            break
```

Comment:
I propose to write with similar approach as below and there is no need to created `src_node` and `dst_node`:
`
            for u, v, d in list(self.node.graph.out_edges(self.node.id, data=True)):
                if d['out'] == self.idx:
                    for key in self.node.graph.get_edge_data(u, v):
                        if self.node.graph.get_edge_data(u, v)[key]['out'] == self.idx:
                            self.node.graph.remove_edge(u, v, key=key)
                            return
`

Reply:
@rkazants I tried it, but there were a few failed unittests. The reason is that your solution changes the disconnect() logic, You want to disconnect the source node from the data node, but I suppose, we should disconnect each destination node from the data node.